(#29, #57) 애플로그인 구현,  Firebase Auth 연동

tngusmiso opened the pull request:
### Related Issue

resolve: #29 
resolve: #57 

### What does this PR do?
- Apple Login Button 터치 시 애플 로그인 모달 띄우기
- Apple 인증 성공 시 Firebase로 한번 더 인증 (nonce 암호화 사용)

### Why are we doing this?
- 소셜로그인이 필요한 앱은 반드시 Apple 로그인까지 구현해야 함
- 여러 소셜로그인 방법으로 가입한 회원을 Firebase Auth로 통합 관리할 수 있음
- 앞으로 구현할 부분
    - 로그인 API 통신 (#58)
    - **실패 시 비회원** => 회원가입 화면으로
    - **성공 시 회원** => 메인 화면으로

### Screenshots
<img src="https://s3.us-west-2.amazonaws.com/secure.notion-static.com/2ae4ac96-7917-47d8-bc84-2ba99430998d/AC0B1C34-3C99-440C-B144-11DED5888AF0.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAT73L2G45O3KS52Y5%2F20210307%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20210307T083013Z&X-Amz-Expires=86400&X-Amz-Signature=708990d309f17f056a4a97f80abb0715c3e136d59207e5065d0e27c88e722b0f&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%22AC0B1C34-3C99-440C-B144-11DED5888AF0.png%22" height="600">



- 한꺼번에 여러 이슈 관련된 PR 올려서 죄송합니다ㅠ0ㅠ

tngusmiso committed:
🚧 (#29) Apple Login 연동

tngusmiso committed:
🚧 (#29) Apple & Firebase Auth 연동

tngusmiso committed:
♻️  (#26) Refactor CustomLoginButton.swift
- label과 imageView를 lazy로 선언하여 생성자 호출 할 때 받아온 값으로 지연 생성하도록 함
- label 폰트 수정
- 생성자에 view를 설정하는 코드와 constraints 설정하는 코드를 함수로 분리함

JeaSungLEE commented on MobalMobal/MobalMobal/Login/CustomLoginButton.swift:
```diff
@@ -8,39 +8,56 @@ import SnapKit
+    
+    lazy var titleLabel: UILabel = {
         let label: UILabel = UILabel()
+        label.font = UIFont(name: "SpoqaHanSansNeo-Medium", size: 14)
```
UIFont도 이곳저곳에서 자주 쓰일거같은데

spoqaHans를 매번 텍스트를 입력하는것보다 UIColor처럼 하나 만들어두면 편하게 슬수 있을거 같습니다.

JeaSungLEE commented on MobalMobal/MobalMobal/Login/CustomLoginButton.swift:
```diff
@@ -8,39 +8,56 @@ import SnapKit
+    
+    // MARK: - Methods
+    private func setupView() {
         self.layer.cornerRadius = self.frame.height / 2
```
우리가 반원 디자인이 많아서
self.roundCorner() 이런거 만들어두면 좋아보입니다.

tngusmiso committed:
🎨 (#29) 반원 뷰를 만드는 메서드 구현 roundCorner()
- Layout Cycle에 맞게 코드 개선
- updateConstraints(): 제약조건 갱신, layoutSubviews(): 레이아웃 갱신

tngusmiso committed:
🔀 (#29) develop branch Merge 반영
- UIColor+Additions.swift : .white20 추가
- UIView+Extension.swift 충돌 해

tngusmiso commented:
* 🎨 리뷰 반영했습니다!
    * Font 관련 사항은 **별도의 이슈**로 빼서 구현했습니다.
* 🔀 `develop` 반영했습니다.

tngusmiso committed:
Merge branch 'develop' into feature/login

tngusmiso committed:
💄 (#26) [로그인] 작은 화면에서 깨지는 UI 수정 - StackView 이용

tngusmiso committed:
💄 (#26) [로그인] Google, Facebook 버튼 순서 수정

tngusmiso commented:
- 작은화면에서 UI가 깨지는 이슈 수정했습니다!
  - `stackView`에 컴포넌트들을 넣고, `view`의 center에 정렬
  - `stackView`에 customSpacing 설정
  - `logoImageView`의 크기는 비율로 조정

    |iPod Touch 7th| iPhone 11 Pro|
    |---|---|
    |<img src="https://s3.us-west-2.amazonaws.com/secure.notion-static.com/73f21f01-b8fe-4c7c-9a36-b8a0847b772d/Untitled.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAT73L2G45O3KS52Y5%2F20210311%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20210311T143152Z&X-Amz-Expires=86400&X-Amz-Signature=ea6c89678b7fabb629ffc9a2bfdc9de37adcd888c640d03afcc8f976f48dd2c9&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%22Untitled.png%22" width = 300>| <img src="https://s3.us-west-2.amazonaws.com/secure.notion-static.com/401e751e-74e8-4a0c-96e1-16091a5aa9bd/66386D06-26C8-4B26-8A3C-F73B972FD46E.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAT73L2G45O3KS52Y5%2F20210311%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20210311T143243Z&X-Amz-Expires=86400&X-Amz-Signature=a460b621ffe65e277d851af0e599265f3093b4fe935ec63aead454d818790ce1&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%2266386D06-26C8-4B26-8A3C-F73B972FD46E.png%22" width = 300>|

- 구글 로그인 버튼과 페이스북 로그인 버튼 위치를 뒤바꿔 놓은 실수까지 수정했습니다!

SongSeoYoung reviewed (approved):
Go!
